### PR TITLE
Use tidyselect::all_of() only if tidyselect >= 1

### PR DESCRIPTION
**Pull request by maurolepore**
This PR avoids a warning due to the ambiguous use an external
vector to select dataframe columns.
    
Using an external vector in selections is ambiguous:
https://tidyselect.r-lib.org/reference/faq-external-vector.html
    
This ambiguity was tolerated in tidyselect < 1. Since then, it
throws a warning:
    
``` r
utils::packageVersion("tidyselect")
#> [1] '1.1.0'

cols <- c("cyl", "disp")
data <- head(mtcars, 3)
dplyr::select(data, cols)
#> Note: Using an external vector in selections is ambiguous.
#> ℹ Use `all_of(cols)` instead of `cols` to silence this message.
#> ℹ See <https://tidyselect.r-lib.org/reference/faq-external-vector.html>.
#> This message is displayed once per session.
#>               cyl disp
#> Mazda RX4       6  160
#> Mazda RX4 Wag   6  160
#> Datsun 710      4  108
```

<sup>Created on 2020-11-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

--

To test this with tidyselect < 1 I used:

* Started from rocker/tidyverse
* installed all dependencies of r2dii.match
* removed tidyselect, dplyr, and tidyr
* installed tidyselect 0.2.5, dplyr 0.8.5, and tidyr 1.0.0

--

Reference: https://principles.tidyverse.org/changes-multivers.html

### CODE REVIEW

File: R/imports.R [hunk 1]
```diff
@@ -7,6 +7,14 @@
 #' @importFrom data.table setDT setkey :=
 NULL
 
+all_of <- function(x) {
+  if (utils::packageVersion("tidyselect") >= "1.0.0") {
+    tidyselect::all_of
+  } else {
+    identity
+  }
+}
```
Comment (maurolepore) [inline R/imports.R:16]: If tidyselect < 1, then `all_of()` does nothing to its input. This means that whenever we support tidyselect 1 (likely via dplyr 1), nothing in the code needs to change -- we could remove this chunk but it won't hurt.

File: R/imports.R [hunk 2]
```diff
@@ -7,6 +7,16 @@
...
 
+# Avoid warning due to ambiguous use of external vector to select df columns:
+# https://github.com/2DegreesInvesting/r2dii.match/pull/317
+all_of_ <- function(x) {
+  if (utils::packageVersion("tidyselect") >= "1.0.0") {
+    tidyselect::all_of
+  } else {
+    identity
+  }
+}
```
Comment (maurolepore) [inline R/imports.R:18]: Here is the thing.

File: R/imports.R [hunk 3]
```diff
@@ -7,6 +7,16 @@
...
 
+# Avoid warning due to ambiguous use of external vector to select df columns:
+# https://github.com/2DegreesInvesting/r2dii.match/pull/317
+all_of_ <- function(x) {
+  if (utils::packageVersion("tidyselect") >= "1.0.0") {
+    tidyselect::all_of(x)
+  } else {
+    identity(x)
+  }
+}
```
Comment (maurolepore) [inline R/imports.R:18]: Here is the thing.

**Review by jdhoffa — APPROVED** [review]
This looks cool, and is a clever/ clean solution!

**Pull request merged**